packages/fxa-auth-server/lib/routes/oauth/index.js

```diff
@@ -317,18 +317,28 @@ module.exports = (log, config, oauthdb, db, mailer, devices) => {
           // (ref: https://bugzilla.mozilla.org/show_bug.cgi?id=1632635), where
           // we need the 'service' event property to distinguish between sync
           // and browser.
-          //
-          // The service for this event gets special-cased to 'sync' so that it
-          // matches its pre-oauth `/certificate/sign` event.  ref:
-          // https://github.com/mozilla/fxa/pull/6581#issuecomment-702248031
           if (
             scopeSet.contains(OAUTH_SCOPE_OLD_SYNC) &&
-            config.oauth.oldSyncClientIds.includes(request.payload.client_id)
+            // To hack around timing issues, only emit this event if it's an
+            // oldsync scope, but not a profile scope. #6578
```

Comment:
I infer from this, that you've observed clients requesting  token with both `oldsync` and `profile` scope. Which clients, and do you know under what circumstances they do so?

In general clients should not be requesting tokens with both of those scopes, for security reasons (e.g. we don't want the profile-server to accidentally be able to read your encrypted sync data, because you used an access_token with additional scopes on it). The only circumstance I can think of where this would be OK, is when initially completing the OAuth login flow by redeeming an authorization code, which returns an access token with all the requested scopes.

Reply:
Great question! Right now, the only spot I've observed both scopes requested together is the auth code exchange step.

Maybe this comment could be reworded. The intent is to document the fact that both fenix and desktop update the account's list of devices before they request a token with the oldsync scope. However, desktop requests a profile token before adding the device to the account, so the right time to increment device usage for all sync clients is when an oldsync token is requested. (This timing assumption seems brittle to me, but also unlikely to change significantly.)

Reply:
@rfk I've updated the code comment. Does it seem clearer?